tests/test_variable.py

```diff
@@ -24,85 +23,83 @@
-        self.assertEqual(str(-p), "Variable 0: name = None,  * -1")
-        self.assertEqual(str(1.2*p*0.4), "Variable 0: name = None,  * 0.48")
-        self.assertEqual(str(1.2*p/2.5), "Variable 0: name = None,  * 0.48")
-
-        def check(par, res):
-            "Apply the parameter mapping, compare with the expected result."
-            temp = np.array([p.val if isinstance(p, Variable) else p for p in par])
-            self.assertAllAlmostEqual(temp, res, self.tol)
-
-        # mapping function must yield the correct parameter values
-        par = [m[k] * Variable(k) for k in range(n)]
-        check(par, par_free * m)
-        par = [Variable(k) * m[k] for k in range(n)]
-        check(par, par_free * m)
-        par = [-Variable(k) for k in range(n)]
-        check(par, -par_free)
-        par = [m[k] * -Variable(k) * m[k] for k in range(n)]
-        check(par, -par_free * m**2)
-
-        # fixed values remain constant
-        check(par_fixed, par_fixed)
-
-    def test_keyword_variable(self):
-        """Keyword Variable reference tests."""
-        self.logTestName()
-
-        n = 10
-        m = nr.randn(n)  # parameter multipliers
-        par_fixed = nr.randn(n)  # fixed parameter values
-        par_free  = nr.randn(n)  # free parameter values
-        Variable.kwarg_values = {k: v for k, v in zip(ascii_lowercase, par_free)}
-
-        # test __str__()
-        p = Variable(0, name='kw1')
-        self.assertEqual(str(p), "Variable 0: name = kw1, ")
-        self.assertEqual(str(-p), "Variable 0: name = kw1,  * -1")
-        self.assertEqual(str(1.2*p*0.4), "Variable 0: name = kw1,  * 0.48")
-        self.assertEqual(str(1.2*p/2.5), "Variable 0: name = kw1,  * 0.48")
-
-        def check(par, res):
-            "Apply the parameter mapping, compare with the expected result."
-            temp = np.array([p.val if isinstance(p, Variable) else p for p in par]).flatten()
-            self.assertAllAlmostEqual(temp, res, self.tol)
-
-        # mapping function must yield the correct parameter values
-        par = [m[k] * Variable(name=n) for k,n in zip(range(n), ascii_lowercase)]
-        check(par, par_free * m)
-        par = [Variable(name=n) * m[k] for k,n in zip(range(n), ascii_lowercase)]
-        check(par, par_free * m)
-        par = [-Variable(name=n) for k,n in zip(range(n), ascii_lowercase)]
-        check(par, -par_free)
-        par = [m[k] * -Variable(name=n) * m[k] for k,n in zip(range(n), ascii_lowercase)]
-        check(par, -par_free * m**2)
-
-        # fixed values remain constant
-        check(par_fixed, par_fixed)
-
-if __name__ == '__main__':
-    print('Testing PennyLane version ' + pennylane.version() + ', Variable class.')
-    # run the tests in this file
-    suite = unittest.TestSuite()
-    for t in (BasicTest,):
-        ttt = unittest.TestLoader().loadTestsFromTestCase(t)
-        suite.addTests(ttt)
-    unittest.TextTestRunner().run(suite)
+
```

Comment:
Since we're refactoring this file, let's add the following to the top to make sure it is deterministic for now:
```suggestion
# make test deterministic
nr.seed(42)
```

Reply:
In a future PR/test refactor, we should probably also aim to remove all random test values, and instead replace them with fixed values.

Reply:
Definitely. But I would be happy to tackle them all together under a new issue.